src/libs/libpng/lv_libpng.c

```diff
@@ -109,6 +110,33 @@ static lv_result_t decoder_info(lv_image_decoder_t * decoder, lv_image_decoder_d
 
         return LV_RESULT_OK;
     }
+    /*If it's a PNG file in a  C array...*/
+    else if(src_type == LV_IMAGE_SRC_VARIABLE) {
+        const lv_image_dsc_t * img_dsc = src;
+        const uint32_t data_size = img_dsc->data_size;
+        const uint32_t * size = ((uint32_t *)img_dsc->data) + 4;
```

Comment:
Let's extract the common code and reuse it for both variable and file.

Reply:
This code was copied from lv_lodepng, so should lv_lodepng be modified in the same way as well?

Reply:
Done